python/docs/index.rst

```diff
@@ -1,11 +1,134 @@
-Apache Qpid Proton: python documentation
-========================================
+####################################
+Qpid Proton Python API Documentation
+####################################
 
-Contents:
+The Proton module provides a Python 2.7 and 3.x API for Qpid Proton. It enables a developer to write Python applications
+that send and receive AMQP messages.
 
+*******
+Modules
+*******
 .. toctree::
    :maxdepth: 2
 
-   tutorial
+   proton
+   proton.handlers
+   proton.reactor
+   proton.utils
+
+*****************************************
+About AMQP and the Qpid Proton Python API
+*****************************************
+
+.. toctree::
+   :maxdepth: 1
+
    overview
+   tutorial
+
+Key API Features
+================
+
+ * Event-driven API
+ * SSL/TLS secured communication
+ * SASL authentication
+ * Automatic reconnect and failover
+ * Seamless conversion between AMQP and Python data types
+ * AMQP 1.0
+
+Basic API Concepts
+==================
+
+The Qpid Python client API and library allows applications to send and receive AMQP messages. See :ref:`overview`
+for a more detailed explanation.
+
+Containers
+----------
+
+Messages are transferred between connected peers (or nodes) using **senders** and **receivers**. Each sender
+or receiver is established over a **connection**. Each connection is established between two unique **containers**,
+the entry point for the API. The container class :class:`proton.reactor.Container` is found in the ``proton.reactor``
+module.
+
+Senders
+-------
+
+The peer that sends messages uses a **sender** to send messages, which includes the target queue or topic which is
+to receive the messages. The sender may be found at :class:`proton.Sender`. Note that senders are most commonly
+obtained by using the convenience method :meth:`proton.reactor.Container.create_sender`.
+
+Receivers
+---------
+
+The peer that receives messages uses a **receiver** to receive messages, and includes a source queue or topic from
+which to receive messages. The receiver may be found at :class:`proton.Receiver`. Note that senders are most commonly
+obtained by using the convenience method :meth:`proton.reactor.Container.create_receiver`.
+
+Message Delivery
+----------------
+
+The process of sending a message is known as **delevery**. Each sent message has a delivry object which tracks the
+status of the message as it is sent from the sender to the receiver. This also includes actions such as settling the
```

Comment:
'settling' is a term of art within the AMQP spec referring to the act of forgetting about a delivery when you no longer need to remember it because we no longer care about it receipt status and no longer need to retry the delivery. It is separate from the concepts of terminal delivery state - Accept, Reject, Release etc.

Reply:
Thank you for the clarification. I changed this to read:

> This also includes actions such as settling the
> delivery (ie causing the delivery status to be forgotten when it is no longer needed).